playbooks/roles/streisand-mirror/tasks/tor-common.yml

```diff
@@ -0,0 +1,42 @@
+---
+
+- name: Make the directory where the Tor Project's mirrored files will be stored
+  file:
+    path: "{{ tor_mirror_location }}"
+    owner: www-data
+    group: www-data
+    mode: 0755
+    state: directory
+
+- name: Discover the latest stable version of the Tor Browser Bundle
+  # The Python code below can be cleaned up once the Tor Project has
+  # finished transitioning away from putting the OS suffix in the
+  # RecommendedTBBVersions file. This check should work with both
+  # formats though.
+  #
+  # https://trac.torproject.org/projects/tor/ticket/8940#comment:28
+  shell: curl -s 'https://www.torproject.org/projects/torbrowser/RecommendedTBBVersions' | python -c 'import json; import re; import sys; j = json.load(sys.stdin); print [re.sub(r"-.*$", "", tbb) for tbb in j if "a" not in tbb and "b" not in tbb][-1];'
```

Comment:
What do you think about refactoring this into a script file copied to & invoked from the host with a `cmd` task? I think it will improve readability and let you do error handling in a more flexible way than as this one-liner within the task.